### PR TITLE
Remove MySQL support

### PR DESCRIPTION
**Pull request by dairbuirabass**
## Description
Remove support for MySQL search engine in the CMP module.

## Motivation and Context
MySQL is not supported as a search engine from Magento 2.4.x onward. Our module has a few extra interceptors for supporting MySQL and it's not worth spending extra time maintaining it for a feature that is sunset in M2. 

## How Has This Been Tested?
Locally.

## Checklist:
- [x] My code follows the code style of this project.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed.
- [x] I have assigned the correct milestone or created one if non-existent.
- [x] I have correctly labeled this pull request.
- [x] I have linked the corresponding issue in this description.
- [x] I have updated the corresponding Jira ticket.
- [ ] I have requested a review from at least 2 reviewers
- [x] I have checked the base branch of this pull request
- [x] I have checked my code for any possible security vulnerabilities

### CODE REVIEW

File: Plugin/Catalog/Block/AbstractBlock.php [hunk 1]
```diff
@@ -91,7 +85,6 @@ abstract class AbstractBlock extends Template
      * @param NostoHelperAccount $nostoHelperAccount
      * @param NostoHelperScope $nostoHelperScope
      * @param StateAwareCategoryServiceInterface $categoryService
-     * @param SearchEngine $searchEngineHelper
      * @param Logger $logger
      */
     public function __construct(
```
Comment (olsi-qose) [inline Plugin/Catalog/Block/AbstractBlock.php:90]: Delete this class. It's not extended by any other class anymore

**Review by olsi-qose — CHANGES_REQUESTED** [review]

**Comment by olsi-qose** [comment]
Delete `ParameterResolverInterface` and `DefaultParameterResolver`. Move `DefaultParameterResolver::getSortingOrder` to the SortOrder.

File: Plugin/Catalog/Block/AbstractBlock.php (hunk 1 above)
Comment (dairbuirabass) [reply Plugin/Catalog/Block/AbstractBlock.php]: Deleted

**Comment by dairbuirabass** [comment]
> Delete `ParameterResolverInterface` and `DefaultParameterResolver`. Move `DefaultParameterResolver::getSortingOrder` to the SortOrder.

@olsi-qose deleted and moved

File: Plugin/Framework/Search/Request/AbstractHandler.php [hunk 2]
```diff
@@ -242,7 +241,7 @@ private function applyCmpFilter(array &$requestData, array $productIds)
         ];
         $requestData['filters']['prod_ids'] = [
             'name' => 'prod_ids',
-            'field' => $this->getProductIdField(),
```
Comment (olsi-qose) [inline Plugin/Framework/Search/Request/AbstractHandler.php:245]: Will this affect versions 2.3.x that still use MySQL?

**Review by olsi-qose — CHANGES_REQUESTED** [review]

**Comment by dairbuirabass** [comment]
@olsi-qose tested with Magento 2.3.7-p2. CMP is ignored completely with MySQL engine.

**Review by olsi-qose — APPROVED** [review]

**Pull request merged**